### PR TITLE
generate SDK for Azure Front Door Service

### PR DESCRIPTION
**Pull request by hytao**
## Description
Genreate .Net SDK for Front Door service.
Swagger: https://github.com/Azure/azure-rest-api-specs/tree/master/specification/frontdoor
PR in sdk-for-net-pr: https://github.com/Azure/azure-sdk-for-net-pr/pull/865

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]

**Comment by dsgouda** [comment]
Please fix the failing tests

**Comment by hytao** [comment]
Hi @dsgouda . Do you know why the test is failing? 
The error message says that "Unable to find recorded mock file", but I do have the recorded .json files checked in.

**Comment by dsgouda** [comment]
@shahabhijeet might be able to help here

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Travis CI is case sensitive and so your directory name where you have saved your recording file needs to match the exact name of your fully qualified name of your test method name.
So in this case the directory name should match the exact name of your namespace name instead of Frontdoor it should be FrontDoor. Making that change will make it find the recorded file.

File: src/SDKs/FrontDoor/AzSdk.RP.props [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--This file and it's contents are updated at build time moving or editing might result in build failure. Take due deligence while editing this file-->
+  <PropertyGroup>
+    <AzureApiTag>Network_2018-08-01;</AzureApiTag>
```
[the remark this answers is not in the document]
Comment (hytao) [reply src/SDKs/FrontDoor/AzSdk.RP.props]: I removed Network_2018-08-01 and build it again. The output are still Network_2018-08-01

File: src/SDKs/FrontDoor/Management.FrontDoor/Microsoft.Azure.Management.FrontDoor.csproj [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<Project Sdk="Microsoft.NET.Sdk">
+	<!-- Please do not move/edit code below this line -->
+	<Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+	<!-- Please do not move/edit code below this line -->
+
+	<PropertyGroup>
+		<PackageId>Microsoft.Azure.Management.FrontDoor</PackageId>
+		<Description>Provides management capabilities for FrontDoor services.</Description>
+		<AssemblyName>Microsoft.Azure.Management.FrontDoor</AssemblyName>
+		<Version>0.1.0</Version>
```
[the remark this answers is not in the document]
Comment (hytao) [reply src/SDKs/FrontDoor/Management.FrontDoor/Microsoft.Azure.Management.FrontDoor.csproj]: done

File: src/SDKs/FrontDoor/FrontDoor.Tests/FrontDoor.Tests.csproj [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <Description>FrontDoor.Tests Class Library</Description>
     <AssemblyName>FrontDoor.Tests</AssemblyName>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
-    <Version>1.0.0</Version>    
+    <Version>0.9.0-preview</Version>
```
Comment (dsgouda) [inline src/SDKs/FrontDoor/FrontDoor.Tests/FrontDoor.Tests.csproj]: Test project should be set to `1.0.0`

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/FrontDoor/FrontDoor.Tests/FrontDoor.Tests.csproj (hunk 3 above)
Comment (hytao) [reply src/SDKs/FrontDoor/FrontDoor.Tests/FrontDoor.Tests.csproj]: How about the version in assemblyinfo.cs (for test project)?

Comment (dsgouda) [reply src/SDKs/FrontDoor/FrontDoor.Tests/FrontDoor.Tests.csproj]: should be `1.0.0.0`

Comment (hytao) [reply src/SDKs/FrontDoor/FrontDoor.Tests/FrontDoor.Tests.csproj]: version updated ,thanks!

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**